charmhelpers/contrib/network/ovs/ovn.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import subprocess
+
+from . import utils
+
+
+def ovn_rundir():
+    """Determine path to OVN sockets.
+
+    Prior to OVN 20.03 the default placement of OVN sockets was together with
+    Open vSwitch, from 20.03 and onwards they are kept in a separate directory.
+
+    :returns: path to OVN rundir
+    :rtype: str
+    """
+    OVN_RUNDIR = '/var/run/ovn'
+    OVS_RUNDIR = '/var/run/openvswitch'
+    for rundir in OVN_RUNDIR, OVS_RUNDIR:
+        if os.path.exists(rundir):
+            return (rundir)
```

Comment:
Not sure why `(rundir)` is in parenthesis?

Reply:
My pre-Python habits present themselves sometimes apparently, fixed.